contracts/TokenHolder.sol

```diff
@@ -392,34 +347,15 @@ contract TokenHolder is MultiSigWallet {
     )
         public
         onlyMultisig
-        keyIsNotNull(_ephemeralKey)
         keyIsAuthorized(_ephemeralKey)
     {
         ephemeralKeys[_ephemeralKey].status = AuthorizationStatus.REVOKED;
     }
 
-    /**
-     * @notice Checks if the specified key is authorized and non-expired.
-     *
-     * @param _ephemeralKey Key to check.
-     *
-     * @return True if the key is currently authorized and has not expired,
-     *         otherwise false.
-     */
-    function isEphemeralKeyActive(address _ephemeralKey)
```

Comment:
We should keep this method as it checks for all the conditions for key to be active and returns true.
Callers can rely on this method to check if key is active or not.
OpenST.js team has already integrated with this method. cc: @kedarchandrayan

Reply:
ok; I had asked @pgev to inline it for improved efficiency on `executeRule`.

We can either keep the function duplicated, but rather we can also perform these checks in openst.js on the public mapping `ephemeralKeys` that is already exposed

Reply:
1. If we bring the function back, we should use it in contract (no code duplication).
2. However, Ben's point is valid, as openst.js can build the function in JS layer.